### PR TITLE
feat(argocd): add argocd deployment lifecycle and summary component

### PR DESCRIPTION
**Pull request by karthikjeeyar**
**Fixes**:

https://issues.redhat.com/browse/RHIDP-1663

This PR changes contains following features/changes

- Add features like Deployment lifecycle and summary components. 
- Removed example components and marking argocd plugin as public.
- Contains dev mode setup to run this plugin in standalone mode. 

**How to test:**

Follow the instructions from the ReadMe file to install and configure the argocd plugin.

**Screenshots:**

image

**Video:**

https://github.com/janus-idp/backstage-plugins/assets/9964343/354614bf-39d9-45f1-adca-2f26975e80a1

**Dev mode:** 
light mode:
Screenshot 2024-04-24 at 12 10 16 AM

image

dark mode:
Screenshot 2024-04-30 at 10 00 11 AM

image

**Unit test coverage:**
image

cc: @rohitkrai03 @invincibleJai

### CODE REVIEW

**Comment by pankajshivpuje** [comment]
@karthikjeeyar Argo CD plug-in will be implemented for the CD tab, correct? Not CI/CD as seen in the screenshot.

**Comment by karthikjeeyar** [comment]
> @karthikjeeyar Argo CD plug-in will be implemented for the CD tab, correct? Not CI/CD as seen in the screenshot.

@pankajshivpuje you are right. It will be mounted to the CD tab in backstage showcase. Since I was using local setup it was named as CI/CD there, I have updated the screenshot to avoid confusions.

File: plugins/argocd/README.md [hunk 1]
```diff
@@ -1,17 +1,145 @@
...
+- To switch between argocd instances, you can use the following annotation
+
+```yaml
+ annotations:
+   ...
+    argocd/instance-name: 'argoInstance2'
+```
+
+**_Note: If this annotation is not set, the plugin will defaultto the first argocd instance configured in the `app.config.yaml`_**
+
```
Comment (rohitkrai03) [inline plugins/argocd/README.md:142]: I think we may also need to add some information about how to use this plugin as dynamic plugin.

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

**Comment by pankajshivpuje** [comment]
@karthikjeeyar I think another think I wanted check is the 'Kind' column in the deployment summary view in the Overview tab. The table is meant to show various kinds of kubernetes resources. Deployment is just one kind. There could be secrets, services or routes etc. Here is board that highlights that thinking. [https://miro.com/app/board/uXjVNGaQ-cI=/?moveToWidget=3458764578942451582&cot=14]

And the last thing I want to mention is server displayed in the card view as well as in the side panel. It may be the information suited for detailed view but I don't have any evidence of where it would work better. We are planning user feedback sessions around this plug-in and based on the feedback we can revisit some of the information we are currently displaying in the card view and the side panel.

Thanks for all the work on this plug-in.

File: plugins/argocd/README.md (hunk 1 above)
Comment (karthikjeeyar) [reply plugins/argocd/README.md:142]: I have updated the `Loading as Dynamic Plugin` section to add more details.

**Review by rohitkrai03 — APPROVED** [review]
/lgtm
/approve

**Comment by invincibleJai** [comment]
Thanks @karthikjeeyar, and one thing that I see is missing is e2e on dev mode.

Will this be covered with follow up PRs

**Comment by karthikjeeyar** [comment]
@invincibleJai e2e on dev mode is being worked on by @jrichter1 , It will be a followup PR.

**Pull request merged**